app/app_proto_conversions.go

```diff
@@ -0,0 +1,630 @@
+		ResourceID:        authorization.ResourceId,
+		IdentityID:        authorization.IdentityId,
+		OrganizationID:    authorization.OrganizationId,
+		IdentityType:      authorization.IdentityType,
+	}
+}
+
+func authorizationToProto(authorization *Authorization) *pb.Authorization {
+	return &pb.Authorization{
+		AuthorizationType: authorization.AuthorizationType,
+		AuthorizationId:   authorization.AuthorizationID,
+		ResourceType:      authorization.ResourceType,
+		ResourceId:        authorization.ResourceID,
+		IdentityId:        authorization.IdentityID,
+		OrganizationId:    authorization.OrganizationID,
+		IdentityType:      authorization.IdentityType,
+	}
+}
+
+// AuthorizedPermissions is authorized permissions.
+type AuthorizedPermissions struct {
+	ResourceType string
+	ResourceID   string
+	Permissions  []string
+}
+
+func authorizedPermissionsFromProto(permissions *pb.AuthorizedPermissions) *AuthorizedPermissions {
+	return &AuthorizedPermissions{
+		ResourceType: permissions.ResourceType,
+		ResourceID:   permissions.ResourceId,
+		Permissions:  permissions.Permissions,
+	}
+}
+
+func authorizedPermissionsToProto(permissions *AuthorizedPermissions) *pb.AuthorizedPermissions {
+	return &pb.AuthorizedPermissions{
+		ResourceType: permissions.ResourceType,
+		ResourceId:   permissions.ResourceID,
+		Permissions:  permissions.Permissions,
+	}
+}
+
+// APIKeyAuthorization is a struct with the necessary authorization data to create an API key.
+type APIKeyAuthorization struct {
+	// `role`` must be "owner" or "operator"
+	role string
+	// `resourceType` must be "organization", "location", or "robot"
+	resourceType string
+	resourceID   string
+}
+
+func createAuthorization(orgID, identityID, identityType, role, resourceType, resourceID string) (*pb.Authorization, error) {
+	if role != "owner" && role != "operator" {
+		return nil, errors.New("role string must be 'owner' or 'operator'")
+	}
+	if resourceType != "organization" && resourceType != "location" && resourceType != "robot" {
+		return nil, errors.New("resourceType must be 'organization', 'location', or 'robot'")
+	}
+
+	return &pb.Authorization{
+		AuthorizationType: role,
+		AuthorizationId:   fmt.Sprintf("%s_%s", resourceType, role),
+		ResourceType:      resourceType,
+		ResourceId:        resourceID,
+		IdentityId:        identityID,
+		OrganizationId:    orgID,
+		IdentityType:      identityType,
+	}, nil
+}
+
+// SharedSecret is a secret used for LocationAuth and RobotParts.
+type SharedSecret struct {
+	ID        string
+	CreatedOn *timestamppb.Timestamp
+	State     SharedSecretState
+}
+
+func sharedSecretFromProto(sharedSecret *pb.SharedSecret) *SharedSecret {
+	return &SharedSecret{
+		ID:        sharedSecret.Id,
+		CreatedOn: sharedSecret.CreatedOn,
+		State:     sharedSecretStateFromProto(sharedSecret.State),
+	}
+}
+
+// SharedSecretState specifies if the secret is enabled, disabled, or unspecified.
+type SharedSecretState int32
```

Comment:
[nit] This and the enums in this PR can just be `int`, I think (`int32` or `int64` depending on your system.) Using `int32` implies some need for a smaller int that probably is not necessary.